include/RAJA/BoxSegment.hxx

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2016, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * All rights reserved.
+ *
+ * This source code cannot be distributed without permission and
+ * further review from Lawrence Livermore National Laboratory.
+ */
+
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file defining box segment classes.
+ *     
+ ******************************************************************************
+ */
+
+#ifndef RAJA_BoxSegment_HXX
+#define RAJA_BoxSegment_HXX
+
+#include "config.hxx"
+
+#include "BaseSegment.hxx"
+
+#include <algorithm>
+#include <iosfwd>
+
+
+namespace RAJA {
+
+
+/*!
+ ******************************************************************************
+ *
+ * \brief  Segment class representing a Box indices.
+ *
+ *         Box is specified by a corner, number of dimensions,
+ *         extent in each dimentsion, and stride to advance in each dimension.
+ *
+ ******************************************************************************
+ */
+class BoxSegment : public BaseSegment
+{
+public:
+
+   ///
+   /// Default box segment ctor.
+   ///
+   /// Segment undefined until begin/end values set.
+   ///
+   BoxSegment()
+   : BaseSegment( _BoxSeg_ ), 
+     m_corner(UndefinedValue), 
+     m_dim(UndefinedValue) { ; }
+
+   ///
+   /// Construct box segment with [begin, end) specified.
+   ///
+   BoxSegment(Index_type corner,  Index_type dim,
+                Index_type const *extent, Index_type const *stride) 
+   : BaseSegment( _BoxSeg_ ) 
+   {
+      m_corner = corner ;
+      m_dim = dim ;
+
+      for (int i=0; i<m_dim; ++i) {
```

Comment:
Since m_extent and m_stride are only length 3, shouldn't there at least be a check here, or switch those to vectors perhaps?